source/server/connection_handler_impl.cc

```diff
@@ -31,8 +31,8 @@ void ConnectionHandlerImpl::addListener(Network::ListenerConfig& config) {
   if (socket_type == Network::Address::SocketType::Stream) {
     listener = std::make_unique<ActiveTcpListener>(*this, config);
   } else {
-    ASSERT(socket_type == Network::Address::SocketType::Datagram,
-           "Only datagram/stream listener supported");
+    RELEASE_ASSERT(config.udpListenerFactory() != nullptr,
```

Comment:
The previous conversation was lost, but this should never happen due to being checked via config elsewhere, right? If so can we make this a normal ASSERT?

Reply:
Early on ListenerImpl() will always create some sort of ActiveUdpListener, if not configured, ActiveRawUdpListener will be created. So yes here ASSERT is sufficient.